Week 6/Subtree/Solution.java

```diff
@@ -0,0 +1,81 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+public class Solution {
+	//Node structure considered for this problem
+	class Node {
+		int value;
+		Node left;
+		Node right;
+
+		Node(int value, Node left,Node right){
+			this.value = value;
+			this.left = left;
+			this.right = right;
+		}
+	}
+
+    public boolean isSubtree(Node t, Node s) {
+    	//input validation
+    	if(t == null) return true;
+    	if(s == null) return false;
+    	//queue for BFS
+    	Queue<Node> bfs = new LinkedList<Node>();
+    	bfs.add(s);
+    	//while there are still nodes on the tree
+    	while(!bfs.isEmpty()){
+    		s = bfs.remove();
+    		//If this node is a leaf ignore it
+    		if(s==null) continue;
+    		//if both t and s have the same value compare
+    		if(t.value == s.value){
+    			//If they have the same children return true
+    			if(checkEquals(t.left,s.left)&&checkEquals(t.right,s.right)){
```

Comment:
nitpick: leave a space on either side of && for better readability and do the same for the other logical operations, ||, == and so on